crates/sparrow-backend/src/pipeline_schedule.rs

```diff
@@ -0,0 +1,163 @@
+use sparrow_core::debug_println;
+use sparrow_physical::{Pipeline, Step, StepKind};
+
+const DEBUG_SCHEDULING: bool = false;
+
+/// Determine the pipeline each step should be part of.
+///
+/// This is computed in a bottom-up manner:
+///
+/// 1. Each leaf (input) is the start of a separate pipeline.
+/// 2. Certain step kinds are "pipeline breaking", which means
+///    they start a new pipeline. For instance, `with_key`.
+/// 3. Any other operation with a single input is part of the
+///    the same pipeline as the input.
+/// 4. Any other operation is a separate pipeline.
+pub fn pipeline_schedule(steps: &[Step]) -> Vec<Pipeline> {
```

Comment:
This is a simple analysis, showing the benefits (simplicity) of using the topologic order.

Reply:
I don't think `pipeline::schedule` makes sense, since I don't think we'd have anything else in that module.

I can see either:

1. `schedule_pipeline` name it a "verb" -- I tend to prefer this for "imperative" things, where it makes sense to describe what the function *does*
2. `pipeline_schedule` name it a "noun phrase" -- something that *describes* what it returns. I tend to prefer this for things that are more "functional" (executed for the result rather than the side-effects).

That said, I'm fine with either `schedule_pipeline` or `pipeline_schedule`.

Reply:
Cool, no makes sense to leave as is then